File: backend-nestjs/src/modules/zones/application/queries/get-all-zones/get-all-zones.handler.ts
import { IQueryHandler, QueryHandler } from '@nestjs/cqrs';
import { GetAllZonesQuery } from './get-all-zones.query';
import { ZoneRepository } from '../../../infrastructure/zone.repository';
import { Zone } from '../../../domain/zone';

@QueryHandler(GetAllZonesQuery)
export class GetAllZonesHandler implements IQueryHandler<
  GetAllZonesQuery,
  Zone[]
> {
  constructor(private readonly repository: ZoneRepository) {}

  async execute(): Promise<Zone[]> {
    return this.repository.findAll();
  }
}

File: backend-springboot/src/main/java/com/greenalgeria/zone/application/query/GetAllZonesHandler.java
package com.greenalgeria.zone.application.query;

import com.greenalgeria.shared.cqrs.QueryHandler;
import com.greenalgeria.zone.application.*;
import com.greenalgeria.zone.domain.ZoneRepository;
import java.util.List;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

@Component
public class GetAllZonesHandler implements QueryHandler<GetAllZonesQuery, List<ZoneResponse>> {

    private final ZoneRepository zoneRepository;

    public GetAllZonesHandler(ZoneRepository zoneRepository) {
        this.zoneRepository = zoneRepository;
    }

    @Transactional(readOnly = true)
    public List<ZoneResponse> handle(GetAllZonesQuery query) {
        return zoneRepository.findAllByOrderByNameAsc().stream()
                .map(ZoneResponse::from)
                .toList();
    }

    @Override
    public Class<GetAllZonesQuery> supportedQuery() {
        return GetAllZonesQuery.class;
    }
}
